internal/pkg/override/noop.go

```diff
@@ -0,0 +1,12 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package override
+
+// Noop represents an Overrider that does not do any transformations.
+type Noop struct{}
```

Comment:
Why do we need this 🤔

Reply:
The idea is to be able to do something like this from a client:

```go
info, err := override.Lookup(path/to/overrides)
if err != nil {
   return new(override.Noop)
}
if info.IsCDK() {
   return override.WithCDK(info.Path())
}
return override.WithYAMLPatch(info.Path())
```

This way the `deploy` package doesn't need to do any  `if override != nil` check and can just call `overrides.Override`

Reply:
Here is a sample scenario to illustrate:
```
copilot/
  frontend/
      manifest.yml
```
There is no `overrides/` directory defined by the customer, so we don't want to error.

My plan is to have a flow like this:
```go
info, err := override.Lookup("copilot/frontend/overrides")
if err != nil {
   if errors.Is(err, override.ErrNotExist) { return new(override.Noop) } // This is fine, have an overrider that doesn't do anythign.
   return err
}
if info.IsCDK() {
  return override.WithCDK(info.Path())
}
return override.WithYAMLPatch(info.Path())
```